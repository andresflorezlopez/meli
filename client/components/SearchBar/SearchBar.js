import React, { useState } from "react";
import PropTypes from "prop-types";
import Container from "@components/Container/Container";
import {
  StyledLogo,
  StyledSearchBar,
  StyledSearchInput,
  StyledSearchButton,
  StyledLogoContainer,
  StyledSearchContainer
} from "./SearchBar.styled";

const SearchBar = ({
  logo,
  placeholder,
  onClickLogo,
  onSearch,
  defaultTerm
}) => {
  const [searchTerm, setSearchTerm] = useState("");
  const onKeyPress = e => {
    e.which === 13 && onSearch(searchTerm);
  };
  const onChange = e => {
    setSearchTerm(e.target.value);
  };
  return (
    <StyledSearchBar>
      <Container direction="row">
        <StyledLogoContainer>
          <StyledLogo
            onClick={() => onClickLogo()}
            src={logo}
            alt="Mercado Libre"
          />
        </StyledLogoContainer>
        <StyledSearchContainer>
          <StyledSearchInput
            placeholder={placeholder}
            onKeyPress={onKeyPress}
            onChange={onChange}
            defaultValue={defaultTerm}
          />
          <StyledSearchButton onClick={() => onSearch(searchTerm)} />
        </StyledSearchContainer>
      </Container>
    </StyledSearchBar>
  );
};

SearchBar.propTypes = {
  logo: PropTypes.string.isRequired,
  onSearch: PropTypes.func.isRequired,
  onClickLogo: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  defaultTerm: PropTypes.string
};

SearchBar.defaultProps = {
  placeholder: "Nunca dejes de buscar",
  defaultTerm: ""
};

export default SearchBar;
