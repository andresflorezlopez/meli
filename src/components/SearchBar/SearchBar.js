import React, { useState } from "react";
import PropTypes from "prop-types";
import Container from "@components/Container/Container";
import {
  StyledLogo,
  StyledSearchBar,
  StyledSearchInput,
  StyledSearchButton,
  StyledLogoContainer,
  StyledSearchContainer
} from "./SearchBar.styled";

const SearchBar = ({ logo, placeholder, onSearch }) => {
  const [searchTerm, setSearchTerm] = useState("");
  const onKeyPress = e => {
    e.which === 13 && onSearch(searchTerm);
  };
  const onChange = e => {
    setSearchTerm(e.target.value);
  };
  return (
    <StyledSearchBar>
      <Container direction="row">
        <StyledLogoContainer>
          <StyledLogo src={logo} />
        </StyledLogoContainer>
        <StyledSearchContainer>
          <StyledSearchInput
            placeholder={placeholder}
            onKeyPress={onKeyPress}
            onChange={onChange}
          />
          <StyledSearchButton onClick={() => onSearch(searchTerm)} />
        </StyledSearchContainer>
      </Container>
    </StyledSearchBar>
  );
};

SearchBar.propTypes = {
  logo: PropTypes.string.isRequired,
  onSearch: PropTypes.func.isRequired,
  placeholder: PropTypes.string
};

SearchBar.defaultProps = {
  placeholder: "Nunca dejes de buscar"
};

export default SearchBar;
