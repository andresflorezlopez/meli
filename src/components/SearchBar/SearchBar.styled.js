import styled from "styled-components";
import searchIcon from "@assets/images/search.png";

export const StyledSearchBar = styled.div`
  background-color: ${({ theme }) => theme.colors.primary};
  padding: ${({ theme }) => theme.spacing(1)} 0;
`;

export const StyledLogoContainer = styled.div`
  align-items: center;
  display: flex;
  flex: 1;
`;

export const StyledLogo = styled.img`
  max-height: 100%;
`;

export const StyledSearchContainer = styled.div`
  position: relative;
  flex: 14;
`;

export const StyledSearchInput = styled.input`
  padding: ${({ theme }) => `${theme.spacing(1)} ${theme.spacing(2)} }`};
  border-radius: ${({ theme }) => theme.borderRadius.normal};
  font-size: ${({ theme }) => theme.fontSize.medium};
  font-family: ${({ theme }) => theme.fontFamily};
  box-sizing: border-box;
  outline: none;
  border: none;
  height: 100%;
  width: 100%;
  ::placeholder {
    color: ${({ theme }) => theme.colors.font.light};
  }
`;

export const StyledSearchButton = styled.div`
  border-radius: ${({ theme }) =>
    `0  ${theme.borderRadius.normal} ${theme.borderRadius.normal} 0`};
  background: url(${searchIcon}), ${({ theme }) => theme.colors.background};
  background-repeat: no-repeat;
  background-position: center;
  position: absolute;
  cursor: pointer;
  height: 100%;
  width: 45px;
  right: 0;
  top: 0;
`;
