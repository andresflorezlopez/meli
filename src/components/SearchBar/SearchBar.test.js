import React from "react";
import SearchBar from "./SearchBar";
import { mount } from "enzyme";

import renderer from "react-test-renderer";
import LightTheme from "@config/themes/light";
import { ThemeProvider } from "styled-components";

describe("SearchBar", () => {
  test("should render", () => {
    const component = renderer.create(
      <ThemeProvider theme={LightTheme}>
        <SearchBar
          logo={"potato.png"}
          placeholder="Nunca dejes de buscar"
          onSearch={value => console.log(value)}
        />
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should call onClick event", () => {
    const spy = jest.fn();
    const wrapper = mount(
      <ThemeProvider theme={LightTheme}>
        <SearchBar
          logo={"potato.png"}
          placeholder="Nunca dejes de buscar"
          onSearch={value => spy(value)}
        />
      </ThemeProvider>
    );

    const input = wrapper.find("input");
    input.simulate("change", { target: { value: "potato" } });
    input.simulate("keyPress", {
      which: 13
    });
    expect(spy).toHaveBeenCalled();
  });
});
