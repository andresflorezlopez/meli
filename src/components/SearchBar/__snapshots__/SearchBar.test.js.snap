// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render 1`] = `
<div
  className="sc-1yye7vx-0 fqiRkl"
>
  <div
    className="sc-1ctswtc-0 kWOxtL"
    direction="row"
  >
    <div
      className="sc-1yye7vx-1 tIPbA"
    >
      <img
        className="sc-1yye7vx-2 dqNuJn"
        src="potato.png"
      />
    </div>
    <div
      className="sc-1yye7vx-3 bhyxsm"
    >
      <input
        className="sc-1yye7vx-4 ixEbfT"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Nunca dejes de buscar"
      />
      <div
        className="sc-1yye7vx-5 jIjCdQ"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
